charts/prometheus-pushgateway/templates/deployment.yaml

```
@@ -25,4 +25,6 @@ spec:
         {{- include "prometheus-pushgateway.defaultLabels" . | nindent 8 }}
     spec:
       {{- include "prometheus-pushgateway.podSpec" . | nindent 6 }}
+      lifecycle:
+        {{- toYaml .Values.lifecycle | nindent 8 }}
```

I'd suggest not to set an empty field:
```suggestion
      {{- with .Values.lifecycle }}
      lifecycle:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

---

Thank you for the feedback.